akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -922,6 +922,35 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
       }
     }
 
+    "enforce settings" should {
+
+      "reject new substreams when exceeding SETTINGS_MAX_CONCURRENT_STREAMS" in new TestSetup with RequestResponseProbes with Http2FrameHpackSupport {
+        // this test exceeds the server setup so we use the server config value as input for the test
+        private val maxStreams: Int = system.settings.config.getInt("akka.http.server.http2.max-concurrent-streams")
```

Comment:
Usually we set a fixed value in the settings (via TestSetup constructor).

Reply:
Yeah, I have another PR built on top of this one with that improvement:
https://github.com/akka/akka-http/pull/3543/files#diff-708fe2ad051369bad850a1616aecc6abb02a09b61d6a2045b1b762cae57a39a0R929-R932